darts/datasets/__init__.py

```diff
@@ -648,3 +648,168 @@ def _to_multi_series(self, series: pd.DataFrame) -> List[TimeSeries]:
+            )
+        ]
+
+
+class TrafficDataset(DatasetLoaderCSV):
+    """
+    The raw data is in http://pems.dot.ca.gov. The data in this repo is a collection of 48 months (2015-2016)
+    hourly data from the California Department of Transportation. The data describes the road occupancy rates (between 0 and 1)
+    measured by different sensors on San Francisco Bay area freeways.
+
+    References
+    ----------
+    .. [1] https://github.com/laiguokun/multivariate-time-series-data
+    """
+
+    def __init__(self, multivariate: bool = True):
+        """
+        Parameters
+        ----------
+        multivariate: bool
+            Whether to return a single multivariate timeseries - if False returns a list of univariate TimeSeries. Default is True.
+        """
+        super().__init__(
+            metadata=DatasetLoaderMetadata(
+                "traffic.csv",
+                uri="https://raw.githubusercontent.com/unit8co/darts/Improvement/Add_new_datasets_617/datasets/traffic.csv",
+                hash="a2105f364ef70aec06c757304833f72a",
+                header_time="Date",
+                format_time="%Y-%m-%d %H:%M:%S",
+                freq="1H",
+                multivariate=multivariate,
+            )
+        )
+
+    def _to_multi_series(self, series: pd.DataFrame) -> List[TimeSeries]:
+        """
+        Load the TrafficDataset dataset as a list of univariate timeseries, one for each ID.
+        """
+        return [
+            series[label]
+            for label in _build_tqdm_iterator(
+                series, verbose=False, total=len(series.columns)
+            )
+        ]
+
+
+class WeatherDataset(DatasetLoaderCSV):
+    """
+    Weather includes 21 indicators of weather, such as air
+    temperature, and humidity. Its data is recorded every
+    10 min for 2020 in Germany.
+
+    References
+    ----------
+    .. [1] https://www.bgc-jena.mpg.de/wetter/
+    .. [2] https://arxiv.org/pdf/2205.13504.pdf
+    """
+
+    def __init__(self, multivariate: bool = True):
+        """
+        Parameters
+        ----------
+        multivariate: bool
+            Whether to return a single multivariate timeseries - if False returns a list of univariate TimeSeries. Default is True.
+        """
+        super().__init__(
+            metadata=DatasetLoaderMetadata(
+                "weather.csv",
+                uri="https://raw.githubusercontent.com/unit8co/darts/Improvement/Add_new_datasets_617/datasets/weather.csv",
+                hash="a2942a05638ba311bc7935bcc087a30f",
+                header_time="Date Time",
+                format_time="%d.%m.%Y %H:%M:%S",
+                freq="10min",
+                multivariate=multivariate,
+            )
+        )
+
+    def _to_multi_series(self, series: pd.DataFrame) -> List[TimeSeries]:
+        """
+        Load the WeatherDataset dataset as a list of univariate timeseries, one for weather indicator.
+        """
+        return [
+            series[label]
```

Comment:
This seems to be returning a pandas Series, we need to return a `TimeSeries`. You can probably simply do `TimeSeries.from_series(series[label])`.